### PR TITLE
fix: error when asset binding is provided without user worker

### PR DESCRIPTION
**Pull request by emily-shen**
## What this PR solves / how to test

Fixes [WC-2744](https://jira.cfdata.org/browse/WC-2744)

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: no e2e's for this yet :(
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Changeset included
  - [ ] Changeset not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: unreleased feat

### CODE REVIEW

File: packages/wrangler/src/experimental-assets.ts [hunk 1]
```diff
@@ -413,6 +420,12 @@ export function verifyMutuallyExclusiveAssetsArgsOrConfig(
 			"Cannot use Experimental Assets and tail consumers in the same Worker. Tail Workers are not yet supported for Workers with assets."
 		);
 	}
+
+	if (!(args.script || config.main) && config.experimental_assets?.binding) {
+		throw new UserError(
+			"Cannot use Experimental Assets with a binding in a assets-only Worker."
```
Comment (petebacondarwin) [inline packages/wrangler/src/experimental-assets.ts:426]:
```suggestion
			"Cannot use Experimental Assets with a binding in an assets-only Worker."
```

**Review by petebacondarwin — APPROVED** [review]
Nice. Just a typo.

File: packages/wrangler/src/experimental-assets.ts (hunk 1 above)
Comment (emily-shen) [reply packages/wrangler/src/experimental-assets.ts]: 😱

File: packages/wrangler/src/experimental-assets.ts [hunk 2]
```diff
@@ -413,6 +420,12 @@ export function verifyMutuallyExclusiveAssetsArgsOrConfig(
 			"Cannot use Experimental Assets and tail consumers in the same Worker. Tail Workers are not yet supported for Workers with assets."
 		);
 	}
+
+	if (!(args.script || config.main) && config.experimental_assets?.binding) {
+		throw new UserError(
+			"Cannot use Experimental Assets with a binding in an assets-only Worker."
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/experimental-assets.ts:426]: We always want to follow up with an action that users can take to solve the error

```suggestion
			"Cannot use Experimental Assets with a binding in an assets-only Worker.\n" +
			`Please specify the path to the entry point for your Worker either in your configuration file ("main"), or as an argument to your wrangler command.`
```

File: packages/wrangler/src/experimental-assets.ts [hunk 3]
```diff
@@ -371,16 +371,23 @@ export function processExperimentalAssetsArg(
 }
 
 /**
+ * Validate assets configuration after CLI args and config have been combined.
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/experimental-assets.ts:374]:
```suggestion
 * Validate assets configuration against the following requirements:
 *     - assets cannot be used in combination with a few other select
 *        Workers features, such as: legacy assets, sites and tail consumers
 *     - an asset binding cannot be used in a Worker that only has assets
 * and throw an appropriate error if invalid.
```

**Review by edmundhung — APPROVED** [review]

File: packages/wrangler/src/experimental-assets.ts [hunk 4]
```diff
@@ -392,7 +398,8 @@ export function verifyMutuallyExclusiveAssetsArgsOrConfig(
 		(args.legacyAssets || config.legacy_assets)
 	) {
 		throw new UserError(
-			"Cannot use Experimental Assets and Legacy Assets in the same Worker."
+			"Cannot use Experimental Assets and Legacy Assets in the same Worker.\n" +
+				"Please remove either the `site` or `experimental_assets` field from your configuration file."
```
[the remark this answers is not in the document]
Comment (emily-shen) [reply packages/wrangler/src/experimental-assets.ts:402]: the wrangler.toml key for sites/legacy assets is `sites`

**Pull request merged**